File: 2021_03_08_old_cs_sln/ImageViewer/ImageViewer/Interface/ILog.vb
Public Interface ILog
    'enableLoging bool 
    Sub showWindow()
    Sub setWindowPosSize(position As Point, size As Size)
    Sub applyWindowPosSize()
    Sub setMode(mode As Integer)
    Sub startLoging()
    Sub stopLoging()
    Sub addLog(value As String)
    Sub initialize(mode As Integer, saveloglevel As Integer, windowVisible As Boolean)
    Sub setFilePath(path As String)
    Sub setFilePath()
    Sub setFileName(Filename As String)
    Sub setFileName()
    Sub initializeLogFile()
    Function getLogFileFullpath() As String

    Sub addLog(priority As Integer, functionName As String, Optional parameter As String = "")
    Function alingmentLog(functionName As String, Optional param As String = "") As String
    Sub addLogMain(value As String)

    Sub addLogToList(value As String)
    Sub addLogToFile(value As String)
    Sub updataWindow(value As String)

    Sub setNotFocus(flag As Boolean)
    Sub updateWindowNotFocus(value As String)
    Sub writeLogAllToFile()
    Sub Dispose()
End Interface

File: 2021_03_08_old_cs_sln/ImageViewer2/PictureBox/Log/ILog.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Drawing;

namespace Log
{
    interface ILog
    {
        int showWindow();
        int setWindowLocationAndSize(Point location, Size size);
        void startLoging();
        void stopLoging();
        void addLog(int priority, string functionName, string parameter = "");
        //void addLogMain(string value);
        void initializeLogFile();
        void addLog(string value);
        void addLogToFile(string value);
        void addLogToList(string value);
        void consoleWriteLine(string str);
        int initialize(int mode, int saveLogLevel, bool windowVisible, string path, string filename);
    }
}
